### PR TITLE
Add recipe for Dlt

### PR DESCRIPTION
**Pull request by sultaniman**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/dlt/meta.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    - fsspec >=2022.4.0
+    - packaging >=21.1
+    - win-precise-time >=1.4.2
+  run_constrained:
+    - psycopg2-binary >=2.9.1
+    - psycopg2cffi >=2.9.0
+    - grpcio >=1.50.0
+    - google-cloud-bigquery-core >=2.26.0
+    - pyarrow >=12.0.0
+    - duckdb >=0.10.0
```
Comment (sultaniman) [inline recipes/dlt/meta.yml:58]: @rudolfix @sh-rp which duckdb version should use?

Comment (rudolfix) [reply recipes/dlt/meta.yml:58]: duckdb is an extra for dlt. let's discuss which extras we want to include in the recipe.

File: recipes/dlt/meta.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    - gitpython >=3.1.29
+    - pytz >=2022.6
+    - giturlparse >=0.10.0
+    - orjson >=3.6.7,<=3.9.10
+    - tenacity >=8.0.2
+    - jsonpath-ng >=1.5.3
+    - fsspec >=2022.4.0
+    - packaging >=21.1
+    - win-precise-time >=1.4.2
+  run_constrained:
```
Comment (xhochy) [inline recipes/dlt/meta.yml:52]: Please remove all the run_constrainted here. This will make the package near to uninstallable.

Comment (sultaniman) [reply recipes/dlt/meta.yml]: Done.

**Comment by sultaniman** [comment]
@xhochy thanks for review I updated the recipe. I am wondering about steps required integrate poetry or will now conda just use dependencies from the list to install via pip?

File: recipes/dlt/meta.yml (hunk 1 above)
Comment (sultaniman) [reply recipes/dlt/meta.yml]: @rudolfix I removed all optional dependencies and updated version

**Comment by xhochy** [comment]
>  I am wondering about steps required integrate poetry or will now conda just use dependencies from the list to install via pip?

`conda` will installed dependencies via `conda` and will not utilise `pip` or `poetry` at all. 

Still, I could only partially parse your question so this might not be the best answer you were looking for.

**Comment by sultaniman** [comment]
@xhochy I meant that `pip` in build was specified, will also be called during build time or only conda will be called?

**Comment by xhochy** [comment]
> @xhochy I meant that `pip` in build was specified, will also be called during build time or only conda will be called?

conda-build calls the script you have specified and that calls `pip`.

**Comment by sultaniman** [comment]
Got it, I wasn't sure if pip will interfaces well with pyproject, just did double check to verify will work out when called by conda builder

**Comment by xhochy** [comment]
`pip` calls the build backend that is specified in the `pyproject.toml`

**Comment by sultaniman** [comment]
There is an error appearing during artifact validation stage
```
The following NEW packages will be INSTALLED:

    _libgcc_mutex:       0.1-conda_forge              conda-forge
    _openmp_mutex:       4.5-2_gnu                    conda-forge
    astunparse:          1.6.3-pyhd8ed1ab_0           conda-forge
    dlt:                 0.4.8-pyhd8ed1ab_0           local      
    typing_extensions:   4.11.0-pyha770c72_0          conda-forge
>>>>tzdata:              2024a-h0c530f3_0             conda-forge
    urllib3:             2.2.1-pyhd8ed1ab_0           conda-forge
    wheel:               0.43.0-pyhd8ed1ab_1          conda-forge
    xz:                  5.2.6-h166bdaf_0             conda-forge
    yaml:                0.2.5-h7f98852_2             conda-forge
import: 'dlt'
import: 'dlt'
+ pip check
dlt 0.4.8 requires tzdata, which is not installed.
WARNING: Tests failed for dlt-0.4.8-pyhd8ed1ab_0.conda - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: dlt-0.4.8-pyhd8ed1ab_0.conda
```
can anyone help me @conda-forge-manager

**Comment by xhochy** [comment]
You probably are looking for `python-tzdata`

**Review by xhochy — APPROVED** [review]

**Comment by sultaniman** [comment]
Thank you @xhochy! 🎉

**Pull request merged**